operator/src/main/java/oracle/kubernetes/operator/helpers/PodStepContext.java

```diff
@@ -599,10 +599,11 @@ protected V1PodSpec createSpec(TuningParameters tuningParameters) {
   }
 
   private List<V1EnvVar> createEnv(V1Container c, TuningParameters tuningParameters) {
-    List<V1EnvVar> envVars = new ArrayList<>();
-    Optional.ofNullable(c.getEnv()).ifPresent(envVars::addAll);
-    envVars.addAll(getEnvironmentVariables(tuningParameters));
-    return envVars;
+    List<V1EnvVar> initContainerEnvVars = new ArrayList<>();
+    Optional.ofNullable(c.getEnv()).ifPresent(initContainerEnvVars::addAll);
+    getEnvironmentVariables(tuningParameters).forEach(envVar ->
+            addDefaultEnvVarIfMissing(initContainerEnvVars, envVar.getName(), envVar.getValue()));
```

Comment:
addDefaultEnvVarIfMissing does not handle V1EnvVar with ValueFrom.

Reply:
As discussed, V1EnvVar with ValueFrom is already handled in getEnvironmentVariables(tuningParameters) method.  I have created a new method `addIfMissing` to add the env variable to list only if it's missing from the list. I also tested setting env var with ValueFrom in init containers and it's handled correctly.

Reply:
can you also verify env var with ValueFrom in server spec are also set correctly in initContainer? Thanks

Reply:
I have pushed a change in 4646578 to handle ValueFrom in `addIfMissing` which will add the env variable with ValueFrom if env value is empty or null. I have also functionally tested that env var is set in initContainer. Thanks.